lib/msal-browser/src/interaction_client/BaseInteractionClient.ts

```diff
@@ -128,6 +132,19 @@ export abstract class BaseInteractionClient {
         return UrlString.getAbsoluteUrl(redirectUri, BrowserUtils.getCurrentUri());
     }
 
+    /*
+     * If authority provided in the request does not match environment/authority specified 
+     * in the account or MSAL config, we throw an error.
+     */
+    async validateRequestAuthority(authority: string, account: AccountInfo): Promise<void> {
+        const discoveredAuthority = await this.getDiscoveredAuthority(authority);
+        const validAuthority = discoveredAuthority.isAlias(account.environment) || discoveredAuthority.isAlias(this.config.auth.authority);
```

Comment:
The check should be `request.authority || config.authority === account.authority)` instead of `request.authority === account.authority || config.authority`